src/lib.rs

```diff
@@ -285,21 +283,6 @@ impl Check {
         self.log_level.as_ref()
     }
 
-    /// Set the termcolor [color choice](termcolor::ColorChoice).
-    /// If `None`, the use of colors will be determined automatically by
-    /// the `CARGO_TERM_COLOR` env var and tty type of output.
-    pub fn with_color_choice(&mut self, choice: Option<termcolor::ColorChoice>) -> &mut Self {
-        self.color_choice = choice;
-        self
-    }
-
-    /// Get the current color choice.  If `None`, the use of colors is determined
-    /// by the `CARGO_TERM_COLOR` env var and whether the output is a tty
-    #[inline]
-    pub fn color_choice(&self) -> Option<&termcolor::ColorChoice> {
-        self.color_choice.as_ref()
-    }
```

Comment:
Any particular reason why we aren't letting library users set the color choice programmatically anymore?

Reply:
We give more preference to the color setting when someone runs a program. Even if someone uses this library, they would have their own color setting mechanism in their binary using `anstyle` and this library would automatically respect it. Therefore, we don't need these settings anymore in the library.

Reply:
What if that third-party program doesn't use `anstyle` for its own styling needs, though? Surely the answer can't be "use `anstyle` or you can't set `cargo-semver-checks` color options" right?

Reply:
> use `anstyle` or you can't set `cargo-semver-checks` color options

I might be completely off here, but I think that's intended and the main reason for `anstyle` repo to exist. They can use `[ColorChoice::write_global](https://docs.rs/colorchoice/latest/colorchoice/enum.ColorChoice.html#method.write_global)` to control the colors in every library they use and do not have to call each individual library's `with_color_choice()`.

And they won't be adding a new dependency, because they already depend on `cargo-semver-checks` which in turn depends on `colorchoice` anyway.

Reply:
You are making a solid case for using `anstyle`. If our downstream user is already directly using `anstyle`, the `ColorChoice::write_global()` approach is what we want them to be able to use.

But I don't see the case for _removing_ the ability to set colors via `with_color_choice()`. If our downstream user is _not_ already directly depending on `anstyle`, it feels wrong to tell them "if you want to control color output in `cargo-semver-checks` then you better add a dependency and configure it there." Even if it isn't a _new_ dependency in the sense of "was it already being compiled," it's still one more thing they have to maintain and keep in sync.

As an example of the burden of needing such a dependency, thus far I've had to deal with more than one case of major version mismatches between a dependency `A`'s dependency `B`, and our directly-added version of `B` that we solely added so we could properly interact with `A`. Let's not impose the same maintenance burden on our users, and let's offer a smooth path for configuring colors even if they don't already use `anstyle`.

Reply:
That makes sense. I didn't think of it like that. My bad.

Reply:
No worries at all! It was a good discussion and `cargo-semver-checks` is better off for it.